lib/engine/round/g_1846/operating.rb

```diff
@@ -118,8 +122,9 @@ def count_actions(type)
 
         def skip_token
           return true if count_actions(Action::PlaceToken).positive?
+          return true unless @current_entity.next_token
 
-          super
+          @current_entity.cash < 40 # Some abilities can get the token down to 40
```

Comment:
I was getting rid of the super call here to avoid the graph.can_token? call. It recomputes the connected_hexes etc. without calling the overrides in round/g_1846/operating.rb. Graph.compute looks up ability[:teleport] which is how the base class gets it to work. 

It either needs to be avoided or I can add an ability[:token] check inside graph#compute. WDYT?

Reply:
i don't think it should be done in compute, i think you can just add teleportation to can_token?

it will actually be more efficient because you can check ability(:token) before the compute and skip the expensive compute if a player can teleport

Reply:
on second thought, i wonder if your code can now be generaliized so we can remove all of the 1846 specific code